goldstone/core/views.py

```diff
@@ -535,41 +535,65 @@ def get(self, request, uuid, *args, **kwargs):
 ###############
 
 # Our API documentation extracts this docstring, hence the use of markup.
-class EventSummarizeView(ElasticListAPIView):
+class EventSummarizeView(DateHistogramAggView):
     """Return an aggregation summary of events from Logstash data.
 
     ---
 
     GET:
         parameters:
+           - name: interval
+             description: The desired time interval, as n(s|m|h|w). E.g., 1d
+                          or 3m.
+             required: true
+             paramType: query
+           - name: per_type
+             description: Include per-type information in the results.
+             type: boolean
+             defaultValue: true
+             paramType: query
            - name: timestamp__range
              description: The time range, as {'xxx':nnn}. Xxx is gte, gt, lte,
                           or lt.  Nnn is an epoch number.  E.g.,
                           {'gte':1430164651890}. You can also use AND, e.g.,
                           {'gte':1430164651890, 'lt':1455160000000}
              paramType: query
-           - name: interval
-             description: The desired time interval, as n(s|m|h|w). E.g., 1d
-                          or 3m.
-             paramType: query
 
     """
 
+    AGG_FIELD = 'timestamp'
+    AGG_NAME = 'per_interval'
     serializer_class = EventAggSerializer
-    reserved_params = ['interval', 'per_host']
+    reserved_params = ['interval', 'per_host', 'per_type']
 
-    class Meta:     # pylint: disable=C1001,W0232
+    class Meta:             # pylint: disable=C1001,W0232,C0111
         model = EventData
 
-    def get(self, request, *args, **kwargs):
+    def get(self, request):
         """Return a response to a GET request."""
 
-        base_queryset = self.filter_queryset(self.get_queryset())
-        interval = self.request.query_params.get('interval', '1d')
+        # Start with a basic histogram search, having a top-level aggregation
+        # for time intervals.
+        search = self._get_search(request)
 
-        data = EventData.ranged_event_agg(base_queryset, interval)
-        serializer = self.serializer_class(data)
+        # See if the request wants per-type information.
+        per_type = \
+            self.request.query_params.get('per_type', 'True')[0].upper() == 'T'
```

Comment:
per_type = 'Twelve' results in per_type = True

how about: 
ast.literal_eval(per_type.capitalize())?  Raises ValueError: malformed string if it isn't a string representation of a [Tt]rue/[Ff]alse.